mmaction/models/recognizers/base.py

```diff
@@ -69,7 +85,8 @@ def __init__(self,
 
     def init_weights(self):
         """Initialize the model network weights."""
-        self.backbone.init_weights()
+        if self.backbone_from in ['mmcls', 'mmaction2']:
+            self.backbone.init_weights()
```

Comment:
What happened for torchvision

Reply:
torchvision init all modules in the __init__ function

Reply:
Users might call this function by hand

Reply:
for which case?

Reply:
You mean users want to init the backbones in their own way?

Reply:
No specific case. I'm concerning the semantic of this function. If the function name is "init weights", then it is expected to init weights. However, current code can silently refuse to init weights based on the value of some external variables.

Reply:
I can add a warning in init_weights of recognizer2d

Reply:
Let's keep this for now. If this is not correct, there will be bug reports eventually.